MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m

```diff
@@ -106,14 +96,53 @@ - (void)applyEnabledState:(BOOL)isEnabled {
 
 - (void)startWithLogManager:(id<MSLogManager>)logManager appSecret:(NSString *)appSecret {
   [super startWithLogManager:logManager appSecret:appSecret];
+  MSLogVerbose([MSUpdates logTag], @"Started Updates service.");
+
+  NSString *updateToken = [MSKeychainUtil stringForKey:kMSUpdateTokenKey];
+  if ([self isEnabled]) {
+    if (updateToken) {
+      self.sender = [[MSDistributionSender alloc]
+          initWithBaseUrl:self.apiUrl
+                  apiPath:[NSString stringWithFormat:kMSUpdtsLatestReleaseApiPathFormat, appSecret]
+                  // TODO: Update token in header should be in format of "Bearer {JWT token}"
+                  headers:@{
+                    kMSHeaderUpdateApiToken : updateToken
+                  }
+             queryStrings:nil
+             reachability:[MS_Reachability reachabilityForInternetConnection]
+           retryIntervals:@[ @(10) ]];
+      [self checkLatestRelease];
+    } else {
+      [self requestUpdateToken];
+    }
+  } else {
+    MSLogDebug([MSUpdates logTag], @"Updates service is disabled, skip update.");
+  }
+}
 
-  // TODO: Hook up with pipeline.
+#pragma mark - Public
+
++ (void)setApiUrl:(NSString *)apiUrl {
+  [[self sharedInstance] setApiUrl:apiUrl];
+}
+
++ (void)setInstallUrl:(NSString *)installUrl {
+  [[self sharedInstance] setInstallUrl:installUrl];
+}
+
++ (void)openUrl:(NSURL *)url {
```

Comment:
This one may be protected by `if ([[self sharedInstance] canBeUsed])` if it's not supposed to be called when the  module is not started yet.